### PR TITLE
gnutls 3.8.0, guile-gnutls 3.7.11 (new formula)

### PR DESCRIPTION
**Pull request by p-linnane**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by p-linnane** [comment]
According to [the release notes](https://lists.gnupg.org/pipermail/gnutls-help/2023-February/004816.html):

>** guile: Guile-bindings removed.
They have been extracted into a separate project to reduce complexity
and to simplify maintenance, see <https://gitlab.com/gnutls/guile/>.

Not sure how we want to handle this.

**Comment by chenrui333** [comment]
not sure why this release is minor version release instead of major version release.

> ** guile: Guile-bindings removed.

sounds like we might want to have a separate formula for `guile-bindings`

**Comment by carlocab** [comment]
> According to [the release notes](https://lists.gnupg.org/pipermail/gnutls-help/2023-February/004816.html):
> 
> > ** guile: Guile-bindings removed.
> > They have been extracted into a separate project to reduce complexity
> > and to simplify maintenance, see https://gitlab.com/gnutls/guile/.
> 
> Not sure how we want to handle this.

This is a huge win, TBH, since it allows us to remove `guile` from the dependency trees of many formulae.

**Comment by carlocab** [comment]
```
  guile-gnutls:
    * GitLab repository not notable enough (<30 forks and <75 stars)
  Error: 1 problem in 1 formula detected
```
Presumably something we can overlook.

**Comment by p-linnane** [comment]
> Presumably something we can overlook.

Definitely, considering the benefits for splitting the formulae out.

File: Formula/guile-gnutls.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                          "--with-guile-site-ccache-dir=#{lib}/guile/3.0/site-ccache",
+                          "--with-guile-extension-dir=#{lib}/guile/3.0/extensions",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules"
+    system "make", "install"
+  end
+
+  def post_install
+    # Touch gnutls.go to avoid Guile recompilation.
+    # See https://github.com/Homebrew/homebrew-core/pull/60307#discussion_r478917491
```
Comment (chenrui333) [inline Formula/guile-gnutls.rb:24]: would this trigger audit failure in the future?

Comment (carlocab) [reply Formula/guile-gnutls.rb:24]: No

File: Formula/gnutls.rb [hunk 2]
```diff
@@ -61,35 +56,16 @@ def install
...
-      If you are going to use the Guile bindings you will need to add the following
-      to your .bashrc or equivalent in order for Guile to find the TLS certificates
-      database:
-        export GUILE_TLS_CERTIFICATE_DIRECTORY=#{pkgetc}/
+      Guile bindings are now in the `guile-gnutls` formula.
     EOS
   end
 
   test do
+    # TODO: Ship a better test.
```
Comment (carlocab) [inline Formula/gnutls.rb:68]: #123082

**Comment by carlocab** [comment]
There are a number of formulae that will need to have an explicit dependency on `readline` and/or `guile`. They currently link to one/both of those through `gnutls`, but we will no longer automatically install `readline` or `guile` after this PR.

These formulae do not need to be rebuilt. (But it would be good to investigate whether we can do without these dependencies...)

**Comment by carlocab** [comment]
> There are a number of formulae that will need to have an explicit dependency on `readline` and/or `guile`. They currently link to one/both of those through `gnutls`, but we will no longer automatically install `readline` or `guile` after this PR.

#123085

**Review by carlocab — APPROVED** [review]
Note: https://github.com/Homebrew/homebrew-core/pull/123085 should be merged before/shortly after this is merged.

**Pull request closed without merging**